compiler/src/Language/Bond/Codegen/Java/Util.hs

```diff
@@ -4,110 +4,63 @@
 {-# LANGUAGE QuasiQuotes, OverloadedStrings, RecordWildCards #-}
 
 module Language.Bond.Codegen.Java.Util
-    ( fieldTypeName
-    , defaultValue
-    , qualifiedName
+    ( qualifiedDeclaredTypeName
     , generatedClassAnnotations
+    , modifierConstantName
+    , isPrimitiveNonEnumBondType
+    , isPrimitiveBondType
+    , isGenericBondStructType
     ) where
 
 import Prelude
-import Data.Int
 import Data.List (intercalate)
-import Data.Text.Lazy (Text, pack)
+import Data.Text.Lazy (Text)
 import Text.Shakespeare.Text
 import Language.Bond.Syntax.Types
-import Language.Bond.Syntax.Util
 import Language.Bond.Codegen.TypeMapping
 import Language.Bond.Codegen.Util()
 
-fieldTypeName :: Type -> Text
-fieldTypeName fieldType = pack $ "com.microsoft.bond.BondDataType." ++ case fieldType of
-    BT_Int8                  -> "BT_INT8"
-    BT_Int16                 -> "BT_INT16"
-    BT_Int32                 -> "BT_INT32"
-    BT_Int64                 -> "BT_INT64"
-    BT_UInt8                 -> "BT_UINT8"
-    BT_UInt16                -> "BT_UINT16"
-    BT_UInt32                -> "BT_UINT32"
-    BT_UInt64                -> "BT_UINT64"
-    BT_Float                 -> "BT_FLOAT"
-    BT_Double                -> "BT_DOUBLE"
-    BT_Bool                  -> "BT_BOOL"
-    BT_String                -> "BT_STRING"
-    BT_WString               -> "BT_WSTRING"
-    BT_Blob                  -> "BT_LIST"
-    BT_Nullable _            -> "BT_LIST"
-    BT_List _                -> "BT_LIST"
-    BT_Vector _              -> "BT_LIST"
-    BT_Set _                 -> "BT_SET"
-    BT_Map _ _               -> "BT_MAP"
-    BT_UserDefined Enum {} _ -> "BT_INT32"
-    BT_UserDefined _ _       -> "BT_STRUCT"
-    -- FIXME: Marker for unsupported types that compiles.
-    _                        -> "BT_UNAVAILABLE"
+-- returns the fully qualified name for a declaration
+qualifiedDeclaredTypeName :: MappingContext -> Declaration -> String
+qualifiedDeclaredTypeName java d@Struct {..}  = intercalate "." $ getDeclNamespace java d ++ [declName]
+qualifiedDeclaredTypeName java d@Enum {..}  = intercalate "." $ getDeclNamespace java d ++ [declName]
+qualifiedDeclaredTypeName java d@Forward {..}  = intercalate "." $ getDeclNamespace java d ++ [declName]
+qualifiedDeclaredTypeName java d@Alias {..}  = intercalate "." $ getDeclNamespace java d ++ [declName]
+qualifiedDeclaredTypeName java d@Service {..}  = intercalate "." $ getDeclNamespace java d ++ [declName]
```

Comment:
Something of a higher-level question - does gbc actually emit services if no rpc mode is enabled? Seems to me like it shouldn't, and Java should error out if a service makes it into Java-specific code before Java supports grpc.